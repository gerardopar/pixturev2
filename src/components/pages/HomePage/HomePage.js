// importing modules
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
// importing redux actions
import { setImagesAsync, setOpenModal, setCloseModal } from '../../../actions/homePageActions';
//importing assets 
import default_bk from '../../../assets/img/bk.jpg';
// importing components
import HeroNav from './Hero/HeroNav';
import HeroTitle from './Hero/HeroTitle';
import HeroSearchBar from './Hero/HeroSearchBar';
import ImageList from '../../UI/ImageList/ImageList';
import ImageModal from '../../UI/ImageModal/ImageModal';
import Spinner from '../../UI/spinner/Spinner';

class HomePage extends Component {
    constructor(props){
        super(props);
    }

    componentWillMount(){
        this.props.setImagesAsync();
    }

    handleImageSearch = (e) => {
        e.preventDefault();
        let imageSearch = e.target.elements.imageSearched.value; // user input
        this.props.history.push(`/search/${imageSearch}?page=1`);
    }

    handleOpenModal = (imgUrl, height, width, user, views, likes) => {
        this.props.setOpenModal(imgUrl, height, width, user, views, likes);
    }

    handleCloseModal = () => {
        this.props.setCloseModal();
    }

    render(){
        return (
            <div>
                <div
                    style={{ backgroundImage: `url(${default_bk})` }}
                    className="hero">
                    <HeroNav />
                    <HeroTitle />
                    <HeroSearchBar 
                        handleImageSearch={this.handleImageSearch}
                    />
                </div>
                {
                    this.props.images.length > 0 ? 
                    <ImageList 
                    handleOpenModal={this.handleOpenModal}
                    images={this.props.images} /> : <Spinner />
                }
                {
                    this.props.modal.hidden 
                    ? null : 
                    <ImageModal 
                        {...this.props.modal}
                        handleCloseModal={this.handleCloseModal}/>
                }
            </div>
        );
    };
};

    // # redux state
    const mapStateToProps = (state) => {
        return {
            images: state.home.images,
            modal: state.home.modal
        }
    };
    
    // # redux actions
    const mapDispatchToProps = (dispatch) => {
        return {
            setImagesAsync: () => dispatch(setImagesAsync()),
            setOpenModal: (imgUrl, height, width, user, views, likes) => dispatch(setOpenModal(imgUrl, height, width, user, views, likes)),
            setCloseModal: () => dispatch(setCloseModal())
        }
    };

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(HomePage));