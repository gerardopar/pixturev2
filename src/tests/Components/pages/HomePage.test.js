import React from 'react';
import { shallow } from 'enzyme';
// importing components
import { HomePage } from '../../../components/pages/HomePage/HomePage';

const homePageProps = {
    modal: {
        img: 'https://imageTesturl.com',
        height: 100,
        hidden: false,
        likes: 1000,
        user: 'http://userTesturl.coom',
        views: 500,
        width: 200
    },
    images: [{ url: 'https://imageTesturl.com' }, { url: 'https://imageTesturl.com' }],
    tag: 'summer'
};

describe('homePage props', () => {
    test('renders homePage component with no props', () => {
        const wrapper = shallow(<HomePage />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
    
    test('renders image modal with expected props', () => {
        const wrapper = shallow(<HomePage modal={homePageProps.modal} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    test('renders category tag with expected props ', () => {
        const wrapper = shallow(<HomePage tag={homePageProps.tag} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    test('renders imageList with expected props ', () => {
        const wrapper = shallow(<HomePage images={homePageProps.images} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
