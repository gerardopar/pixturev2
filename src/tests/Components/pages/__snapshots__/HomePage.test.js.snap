// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homePage props renders category tag with expected props  1`] = `
"<div data-test=\\"component-homepage\\">
  <hero handleImageSearch={[Function]} />
  <categoryTag tag=\\"summer\\" />
  <spinner />
  <imageModal handleCloseModal={[Function]} height={0} img=\\"\\" likes={0} user=\\"\\" views={0} width={0} />
</div>"
`;

exports[`homePage props renders homePage component with no props 1`] = `
"<div data-test=\\"component-homepage\\">
  <hero handleImageSearch={[Function]} />
  <categoryTag tag=\\"\\" />
  <spinner />
  <imageModal handleCloseModal={[Function]} height={0} img=\\"\\" likes={0} user=\\"\\" views={0} width={0} />
</div>"
`;

exports[`homePage props renders image modal with expected props 1`] = `
"<div data-test=\\"component-homepage\\">
  <hero handleImageSearch={[Function]} />
  <categoryTag tag=\\"\\" />
  <spinner />
  <imageModal img=\\"https://imageTesturl.com\\" height={100} hidden={false} likes={1000} user=\\"http://userTesturl.coom\\" views={500} width={200} handleCloseModal={[Function]} />
</div>"
`;

exports[`homePage props renders imageList with expected props  1`] = `
"<div data-test=\\"component-homepage\\">
  <hero handleImageSearch={[Function]} />
  <categoryTag tag=\\"\\" />
  <imageList handleOpenModal={[Function]} images={{...}} />
  <imageModal handleCloseModal={[Function]} height={0} img=\\"\\" likes={0} user=\\"\\" views={0} width={0} />
</div>"
`;
